### PR TITLE
FIX: Publish button correct state for Unpublished records

### PR DESCRIPTION
**Pull request by christopherdarling**
Fixes https://github.com/unclecheese/silverstripe-gridfield-betterbuttons/issues/65

Uses pretty much the same logic found in SiteTree

### CODE REVIEW

**Comment by unclecheese** [comment]
Thank you. @stevie-mayhew any thoughts on this?

File: code/extensions/GridFieldBetterButtonsItemRequest.php [hunk 1]
```diff
@@ -548,11 +548,35 @@ public function recordIsPublished() {
...
 
+	/**
+	 * Determines if the current record is deleted from stage
+	 * @returns boolean
+	 */
+	public function recordIsDeletedFromStage() {
+
+		// for SiteTree records
+		if ( $this->owner->hasMethod('getIsDeletedFromStage') ) {
+			return $this->IsDeletedFromStage;
```
Comment (stevie-mayhew) [inline code/extensions/GridFieldBetterButtonsItemRequest.php]: Should this call the function `$this->owner->IsDeletedFromStage()`

File: code/extensions/GridFieldBetterButtonsItemRequest.php [hunk 2]
```diff
@@ -548,11 +548,35 @@ public function recordIsPublished() {
...
+		}
+
+		if(!$this->owner->record->checkVersioned()) return false;
```
Comment (stevie-mayhew) [inline code/extensions/GridFieldBetterButtonsItemRequest.php]: I'm not sure what @unclecheese prefers but I like all if statements to have braces on them.

File: code/extensions/GridFieldBetterButtonsItemRequest.php [hunk 3]
```diff
@@ -548,11 +548,35 @@ public function recordIsPublished() {
...
+		if (!$this->owner->record->isInDB()) {
+			return true;
+		}
+
+		$table = $this->owner->record->class;
```
Comment (stevie-mayhew) [inline code/extensions/GridFieldBetterButtonsItemRequest.php]: Should this be called $class - minor but then it matches what you are retrieving from the record

**Comment by stevie-mayhew** [comment]
Just one issue with calling a property on the class which won't exist and a couple of style things.

**Comment by christopherdarling** [comment]
Good spots @stevie-mayhew - done those changes

**Comment by unclecheese** [comment]
Thanks, heaps, @stevie-mayhew for the review. Awesome stuff, @christopherdarling !

**Pull request merged**